### ISSUE TITLE
[Customer][Order->Edit] Inconsistent table format on (editable) order confirmation page

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#10261]
## Description

For editable orders - i.e., orders which can be cancelled or line items amount changed by the customer, while the OC is open - the table in the order confirmation page shows an inconsistent format.

## Expected Behavior

Table format should be consistent.

## Actual Behaviour

Table formatting is inconsistent (blank space is visible).

## Steps to Reproduce

1. As an admin: Set up a hub to accept order edits - Under Shop Preferences toggle "Customers can change / cancel orders while order cycle is open"
2. As a customer: Place an order on that Hub
3. As a customer: Edit that order - visit `/account#/orders` and click `Edit` for that order -> Notice the blank space on the table.

(This affects only order confirmation page while the OC is open:
4. As an admin: Close the order cycle
5. As a customer: Edit that order -> notice the formatting is correct)

OR:

Skip all steps and visit:
`https://staging.openfoodnetwork.org.uk/orders/R632653401?order_token=a629d72866356506`

## Animated Gif/Screenshot

![image](https://user-images.githubusercontent.com/49817236/211863635-0690e69b-deab-4531-b3c3-665ef74f3609.png)

## Workaround

Does not affect functionality.

## Severity

bug-s5: we can live with it, only a few users impacted

## Your Environment

* Version used: v4.2.29
* Browser name and version: Firefox 108.02 (does not affect Chrome)
* Operating System and version (desktop or mobile): Ubuntu 22.04

**Comment by drummer83** [comment]
Maybe introduced when we added 'Back to Website' button (not visible on your screenshot)?

**Comment by jibees** [comment]
I'm not sure to know how to reproduce this one... 
by `As a customer: Edit that order`, what do you mean?
This is what I get:

<img width="1225" alt="Capture d’écran 2023-01-12 à 09 26 35" src="https://user-images.githubusercontent.com/296452/212015793-8ca03e44-f7be-47af-b8a5-23627ab6bb9b.png">

**Comment by filipefurtad0** [comment]
Hey @jibees ,

Sorry, I guess I should have been more explicit :see_no_evil: I've edited above.
If the enterprise is set accordingly (step 1) and the order cycle is still open then, the customers can edit orders. Do to so, log in as a customer and visit `/account#/orders` and click `Edit` for that order:

![image](https://user-images.githubusercontent.com/49817236/212041174-92d8dc06-b0f5-47b6-9941-806b0512a492.png)

This will lead the user to the order confirmation screen.

**Comment by jibees** [comment]
Found it. 
This is due to https://github.com/openfoodfoundation/openfoodnetwork/pull/10103
and that particular line `#14`: 

https://github.com/openfoodfoundation/openfoodnetwork/blob/fd446970cafdaf311844b46a3299fb1de3141715/app/webpacker/css/darkswarm/cart-page.scss#L12-L14

the `<table />` is therefore `display: block;` instead of its inherit display which is `display: table;`

**Comment by RachL** [comment]
FYI this is reproductible on order confirmation page as well. I've just tested on production:

![image](https://user-images.githubusercontent.com/6525576/212292421-dfa56a02-8ded-4d7c-8f6c-b9010b1d26c3.png)

**Comment by filipefurtad0** [comment]
> FYI this is reproductible on order confirmation page as well. I've just tested on production:

Yes, I can see it too in order confirmation pages (either editing or after checkout), but only for editable orders. Is that the case, on that pic @RachL ?

**Comment by RachL** [comment]
Yep!

**Comment by ashwini-seshadri** [comment]
Hello, is this issue still open? If so, I would love to work on this! :) Thank you!

**Comment by sigmundpetersen** [comment]
Sure, thank you @ashwini-seshadri 👍

**Comment by drummer83** [comment]
Thank you @ashwini-seshadri!
I have just checked. The problem is still there, but it's smaller now - at least on my device and browser (Firefox).

Here is a screenshot:
![image](https://github.com/user-attachments/assets/9e752012-76f3-4cd8-b4ba-9ea576fcb505)

**Comment by sigmundpetersen** [comment]
Sure go ahead @ndossougbete 👍 
I'll assign you. Thanks!

**Comment by ndossougbete** [comment]
I took some time to look into it today, and I need to correct the statement in my comment above. I think it will be more complicated. The fix added in PR #10103 to address issue #9976 does not seem to be working anymore. 

Repro steps:
  1. A mobile device or some way to simulate it will be needed. I used the mobile/responsive mode from dev tools. E.g. on Firefox , the Galaxy Note 20 at 412px width.
  2. As an admin: In one of the hubs, edit some of the items to make sure a few have some reasonably long name without spaces, so that it is not auto-wrapped. Example: on my simulated phone above, I observed "Mushrooms" to be long enough to cause the issue.
  3. As a customer, on a mobile device: Edit an already placed order:
     - Place an order on that hub
      - Visit `/account#/orders` and click "Edit" for that order -> Notice that the whole page scrolls horizontally

One one hand I don't quite understand why the change to make the `table` have `display: block` does not work, but on the other hand I think it's not going to be the right solution. It is [mentioned on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table), but with a caveat:

 > you'll notice that table's `display` property has been set to `block`. While this allows scrolling, the table loses some of its integrity, and table cells try to become as small as possible.

Chrome seems to handle the "cells try to become as small as possible" part better than Firefox, which is why a gap can show up. I found an article mentioning that it might cause [some accessiblity issues](https://www.tpgi.com/short-note-on-what-css-display-properties-do-to-table-semantics/). And overall, the whole body still scrolls on mobile Chrome & Firefox when a long string without spaces is included in the table. On Firefox is extra bad, compare outcome of executing the snippet here: https://stackoverflow.com/a/48736763. Other difference, in Chrome the "Back to store" button seems to have the wrong size, while in Firefox it matches the size of the buttons below.
![Screenshot from 2024-11-11 21-07-47](https://github.com/user-attachments/assets/983ca7b5-15f0-4857-951d-bd0d9d10220a)

I'll have another look when I have some time. Hopefully I missed some simple fix, but maybe it will take a more comprehensive rewrite of the markup to fix this. Maybe CSS-`calc`'ing the columns get elements with the right sizes?

All that to say that I'll hold off of submitting a PR just removing the `display: block` properties, it seems irrelevant to do considering the other issues, and even makes things worse in certain cases :sweat_smile: 

And please let me know if I should fork the general table/scrolling issues of these screens to a separate issue.

**Comment by ndossougbete** [comment]
Hi! I gave up on trying to get the table to work on mobile, and created PR #13010 to fix the bug initially mentioned here.

I think the table will never work properly on mobile and it would be best to try an alternate layout to display this info without needing to scroll.  An interpretation of this (that still needs a lot of work to be anything close to shippable) could be something like:
![Screenshot from 2024-12-01 16-48-17](https://github.com/user-attachments/assets/1049630f-0e45-47a5-8f76-ec9cce8469fb).
It removes some columns and attempts to put the info back in the first cell instead. But that's becoming off-topic, I think the mobile tables should be forked to a separate issue (or reopen #9976 maybe?) if this is to be done.

**Comment by sigmundpetersen** [comment]
@openfoodfoundation/reviewers @RachL do we want to finish off the work done in the connected PR?
Or should we just move this back to All the things?

**Comment by RachL** [comment]
@sigmundpetersen I'm afraid we don't have budget for this currently :(

**Comment by guidoDutra** [comment]
Hello! I contributed with a few PR some time ago, and would like to contribute some more now. Can this task be assigned to me?
